parser.y

```diff
@@ -2932,7 +2939,10 @@ ConstraintElem:
 		if $7 != nil {
 			c.Option = $7.(*ast.IndexOption)
 		}
-		c.Name = $3.([]interface{})[0].(string)
+		indexName := $3.([]interface{})[0].(*ast.NullString)
+		c.IfEmptyIndex = !indexName.Valid
```

Comment:
These two assignments can put into initialization.

Reply:
Is it not good to read as follows？

```
c := &ast.Constraint{
			Tp:   ast.ConstraintPrimaryKey,
			Keys: $5.([]*ast.IndexPartSpecification),
			Name: $3.([]interface{})[0].(*ast.NullString).String,
			IsEmptyIndex:!$3.([]interface{})[0].(*ast.NullString).Valid,
		}
```

Reply:
fixed